WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Package Selection/Package Creation/ShippingLabelCustomPackageFormViewModel.swift

```diff
@@ -35,25 +37,92 @@ final class ShippingLabelCustomPackageFormViewModel: ObservableObject {
     ///
     @Published var emptyPackageWeight: String
 
-    init(lengthUnit: String? = ServiceLocator.shippingSettingsService.dimensionUnit,
-         weightUnit: String? = ServiceLocator.shippingSettingsService.weightUnit,
-         packageName: String = "",
-         packageType: PackageType = .box,
-         packageLength: String = "",
-         packageWidth: String = "",
-         packageHeight: String = "",
-         emptyPackageWeight: String = "") {
+    /// Validated custom package
+    ///
+    private var validatedCustomPackage: ShippingLabelCustomPackage? {
+        guard isPackageValidated, let boxWeight = Double(emptyPackageWeight) else {
+            return nil
+        }
+        let isLetter = packageType == .letter
+        let dimensions = "\(packageLength) x \(packageWidth) x \(packageHeight)"
+        return ShippingLabelCustomPackage(isUserDefined: true,
+                                          title: packageName,
+                                          isLetter: isLetter,
+                                          dimensions: dimensions,
+                                          boxWeight: boxWeight,
+                                          maxWeight: 0)
+    }
+
+    // MARK: Validation Properties & Publishers
+
+    var isNameValidated = true
+    var isLengthValidated = true
+    var isWidthValidated = true
+    var isHeightValidated = true
+    var isWeightValidated = true
+    var isPackageValidated = false
+
+    lazy var packageNameValidation: ValidationPublisher = {
+        $packageName.hasContent()
+    }()
+
+    lazy var packageLengthValidation: ValidationPublisher = {
+        $packageLength.greaterThan(0)
+    }()
+
+    lazy var packageWidthValidation: ValidationPublisher = {
+        $packageWidth.greaterThan(0)
+    }()
+
+    lazy var packageHeightValidation: ValidationPublisher = {
+        $packageHeight.greaterThan(0)
+    }()
+
+    lazy var packageWeightValidation: ValidationPublisher = {
+        $emptyPackageWeight.greaterThanOrEqualTo(0)
+    }()
```

Comment:
I think this implementation along with the use of `onChange` are more complicated than necessary. We can simplify the logic like so:
- For package name validation, we can make `isNameValidated` a  publisher. We keep the value `true` by default to not show any validation error initially.
```Swift
@Published private(set) var isNameValidated: Bool = true
```
- Then after intializing all stored properties, we can configure the value of this publisher by observing the value of package name starting from the 2nd value (using `dropFirst`):
```Swift
$packageName
        .dropFirst()
        .map { $0.trimmingCharacters(in: .whitespacesAndNewlines).isNotEmpty }
        .assign(to: &$isNameValidated)
```
This way we can omit both `packageWeightValidation` and `onChange`, while keeping the same behavior - starting validation only from the 2nd value. We can apply the same thing to other fields as well.

What makes me prefer this solution: 
- It's simpler and more straightforward, and no logic is hidden in other encapsulation like in `ValidationPublishers`.
- Writing unit tests for `@Published` variables is so much easier since we can access their values directly without having to use `.sink` for observing.

Reply:
Thank you so much! This simpler solution makes a lot of sense to me. I was trying to come up with something like this but couldn't figure out how to make it work, and I think I was just missing the fact that I could use `.dropFirst()` directly when observing the value to get what I wanted (rather than using it in the abstracted publisher). I'll work on making the changes to use this approach. 👍